packages/admin-scripts/src/application/script-runner.js

```diff
@@ -0,0 +1,254 @@
+     */
+    async execute(scriptName, params = {}, options = {}) {
+        const { trigger = 'MANUAL', audit = {}, executionId: existingExecutionId, dryRun = false } = options;
+
+        // Get script class
+        const scriptClass = this.scriptFactory.get(scriptName);
+        const definition = scriptClass.Definition;
+
+        // Validate integrationFactory requirement
+        if (definition.config?.requiresIntegrationFactory && !this.integrationFactory) {
+            throw new Error(
+                `Script "${scriptName}" requires integrationFactory but none was provided`
+            );
+        }
+
+        let executionId = existingExecutionId;
+
+        // Create execution record if not provided
+        if (!executionId) {
+            const execution = await this.commands.createScriptExecution({
+                scriptName,
+                scriptVersion: definition.version,
+                trigger,
+                mode: options.mode || 'async',
+                input: params,
+                audit,
+            });
+            executionId = execution.id;
+        }
+
+        const startTime = new Date();
+
+        try {
+            // Update status to RUNNING (skip in dry-run)
+            if (!dryRun) {
+                await this.commands.updateScriptExecutionStatus(executionId, 'RUNNING');
+            }
+
+            // Create frigg commands for the script
+            let frigg;
+            let operationLog = [];
+
+            if (dryRun) {
+                // Dry-run mode: wrap commands to intercept writes
+                frigg = this.createDryRunFriggCommands(operationLog);
+            } else {
+                // Normal mode: create real commands
+                frigg = createAdminFriggCommands({
+                    executionId,
+                    integrationFactory: this.integrationFactory,
+                });
+            }
+
+            // Create script instance
+            const script = this.scriptFactory.createInstance(scriptName, {
+                executionId,
+                integrationFactory: this.integrationFactory,
+            });
+
+            // Execute the script
+            const output = await script.execute(frigg, params);
+
+            // Calculate metrics
+            const endTime = new Date();
+            const durationMs = endTime - startTime;
+
+            // Complete execution (skip in dry-run)
+            if (!dryRun) {
+                await this.commands.completeScriptExecution(executionId, {
+                    status: 'COMPLETED',
+                    output,
+                    metrics: {
+                        startTime: startTime.toISOString(),
+                        endTime: endTime.toISOString(),
+                        durationMs,
+                    },
+                });
+            }
+
+            // Return dry-run preview if in dry-run mode
+            if (dryRun) {
```

Comment:
mixing the prod logic with dryrun logic will end up bloating the code unnecessarily. I would be happier if we could have an app runner specific for dryrun and another one for prod, just like we do for the repositories for instance.